### PR TITLE
Move LockableCurrency trait to fungible::Lockable and deprecate LockableCurrency

### PR DESCRIPTION
**Pull request by tonyalaribe**
Priliminary efforts to introduce a Lockable trait based off LockableCurrency.

This moves us closer to the points in https://github.com/paritytech/substrate/issues/8453

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2140867

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2140928

File: frame/balances/src/tests.rs [hunk 1]
```diff
@@ -28,15 +28,15 @@ macro_rules! decl_tests {
 		use frame_support::{
 			assert_noop, assert_storage_noop, assert_ok, assert_err,
 			traits::{
-				LockableCurrency, LockIdentifier, WithdrawReasons,
+				fungible, fungible::Lockable, WithdrawReasons,
```
Comment (jsidorenko) [inline frame/balances/src/tests.rs:31]: What would happen if you don't include `fungible::Lockable`?

Comment (tonyalaribe) [reply frame/balances/src/tests.rs:31]: I get this, if fungible::Lockable isn't included:
image

Comment (jsidorenko) [reply frame/balances/src/tests.rs:31]: ouch, got it :)

**Comment by kianenigma** [comment]
How does this work link to https://github.com/paritytech/substrate/issues/12918? has it been discussed/coordinated?

File: frame/support/src/traits.rs [hunk 2]
```diff
@@ -20,6 +20,7 @@
 //! NOTE: If you're looking for `parameter_types`, it has moved in to the top-level module.
 
 pub mod tokens;
+#[allow(deprecated)]
```
Comment (melekes) [inline frame/support/src/traits.rs:23]: do we really need this?

**Review by melekes — APPROVED** [review]
👍 (code-wise; don't know enough to reason about business logic)

**Comment by tonyalaribe** [comment]
> How does this work link to https://github.com/paritytech/substrate/issues/12918? has it been discussed/coordinated?

It's intended to be a step in that direction `moving LockableCurrency to fungibles::*` but given the work in https://github.com/paritytech/substrate/pull/12951, simply renaming LockableCurrency is not a good enough solution. So i'll close this PR for now.

**Pull request closed without merging**